File: Python/ex02.py
def swap(x, y):
    """
    On peux aussi utiliser une variable temporaire pour changer les valeurs (3 étapes)
    Ou swapper les valeurs  nativement en python
    ex 1 :
    temp = x
    x = y
    y = temp
    ex 2 :
    x, y = y, x
    """
    # Renvoie des valeurs changer
    return y, x


# ask x and y
x = int(input("x ?"))
y = int(input("y ?"))

# before swap
print(f"x : {x}, y : {y}")
x, y = swap(x, y)
# after swap
print(f"x : {x}, y : {y}")

File: C/ex02.c
#include <stdio.h>

void swap(int *x, int *y);


int main(int argc, char *argv[])
{
    int x;
    int y;

    printf("x ? ");
    scanf("%i", &x);

    printf("y ? ");
    scanf("%i", &y);

    // print values before swap
    printf("x : %i et y : %i\n", x, y);
    swap(&x, &y);
    // print values after swap
    printf("x : %i et y : %i\n", x, y);
}

void swap(int *x, int *y)
{
    // change values with address
    int temp = *x;
    *x = *y;
    *y = temp;
}